kernel/src/arch/x86/smp/smp.cpp

```diff
@@ -177,163 +177,90 @@ namespace smp
-                                                        '7',
-                                                        '8',
-                                                        '9',
-                                                        '-',
-                                                        '4',
-                                                        '5',
-                                                        '6',
-                                                        '+',
-                                                        '1',
-                                                        '2',
-                                                        '3',
-                                                        '0',
-                                                        '.',
-                                                        /* three gaps */ 0,
-                                                        0,
-                                                        0,
-                                                        /* F11, F12 */ 0,
-                                                        0};
-                        /* 0x1B = ESCAPE; 0x08 = BACKSPACE; 0x09 = HORIZONAL TAB, used to represent TAB
-                        * 0x0D = CARRIAGE RETURN, used to represent ENTER
-                        * 0x11 = DEVICE CONTROL 1, used to represent LEFT CONTROL; 0x0E = SHIFT OUT, used to represent LEFT SHIFT
-                        * 0x0F = SHIFT IN, used to represent RIGHT SHIFT; 0x13 = DEVICE CONTROL 3, used to represent LEFT ALT
-                        * 0x02 = START OF TEXT, used to represent CAPS LOCK
-                        * TODO: Handle F1-F12, Numlock and Scroll Lock, multimedia keys
-                        // also Home, Page Up, etc
-                        * TODO eventually: distinguish Keypad / (0xE0, 0x35) and enter (0xE0, 0x1C) from normal, and RIGHT CTRL and ALT from LEFT
-                      */
-                        uint8_t decoded_keypress;
-                      bool is_pressed;
-                      static constexpr auto FLAG_RELEASED= 0x80;
-                        std::tie (decoded_keypress, is_pressed} = decode_keyboard_scancode(encoded_keypress, us_qwerty_translation, FLAG_RELEASED)
-                        if (decoded_keypress != 0)
+            if (core_id == 0)
+            {
+                proc::make_kthread_args(
+                    +[](std::uint64_t arg) {
+                        klog::log("I got an argument and I don't know what it's for but here it is: %lu", arg);
+                        auto prev_msg = 0;
+                        while (true)
                         {
-                            // THIS IS A TEMPORARY SWITCH CASE FOR DEBUGGING, TODO: DELETE IT AND ACTUALLY HANDLE STUFF
-                            char output_str[15] = "";
-                            switch (decoded_keypress)
+                            auto encoded_keypress = inb(ioports::KEYBOARD);
```

Comment:
Again, this code should be moved to ps2.cpp 
Nothing should be in smp.cpp except for
while(true) { poll_key(); }

Reply:
the issue with that is it makes preventing duplicate presses/signals difficult. Once we have interrupts we can do that, unless you want the while loop in the function as well.